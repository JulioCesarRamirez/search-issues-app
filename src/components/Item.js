import React from 'react';
import PropTypes from 'prop-types';

export const Item = ({ title, url }) => {
  return (
    <div className="card">
      <a href={url} target="_blank">
        {title}
      </a>
    </div>
  );
};

Item.prototype = {
  issue: PropTypes.string.isRequired,
};
