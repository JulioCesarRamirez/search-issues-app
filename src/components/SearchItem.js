import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchItem = ({ setItemToSearch }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setItemToSearch((items) => [inputValue, ...items]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange}/>
    </form>
  );
};

SearchItem.propTypes = {
  setItemToSearch: PropTypes.func.isRequired,
};
