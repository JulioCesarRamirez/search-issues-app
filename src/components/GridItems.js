import React from 'react';
import PropTypes from "prop-types";

import { useFetch } from '../hooks/useFetch';
import { Item } from './Item';

export const GridItems = ({ item }) => {
  const { data: itemsFromApi } = useFetch(item);
  return (
    <div className="card-grid">
      {itemsFromApi.map((item, i) => (
        <Item key={i} {...item} />
      ))}
    </div>
  );
};

GridItems.protoTypes = {
  item: PropTypes.string.isRequired,
};
