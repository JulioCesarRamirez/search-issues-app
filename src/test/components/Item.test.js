import React from 'react';
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { Item } from '../../components/Item';
configure({ adapter: new Adapter() });

describe('Test in <Item />', () => {
  const title = 'A title';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<Item title={title} url={url} />);

  it('Should show <Item /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should has a paragraph with the title', () => {
    const a = wrapper.find('a');
    expect(a.text().trim()).toBe(title);
  });
  it('Should has an image similar to url and alt from props', () => {
    const link = wrapper.find('a');
    expect(link.prop('href')).toBe(url);
  });
  it('Should has card className', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('card')).toBe(true);
  });
});
