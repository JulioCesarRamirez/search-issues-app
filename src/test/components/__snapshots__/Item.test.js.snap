// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Item /> Should show <Item /> correctly 1`] = `ShallowWrapper {}`;
